lua/core/mappings.lua

```diff
@@ -96,73 +96,79 @@ M.comment = function()
    map("v", "<leader>/", "<esc><cmd> :lua require('Comment.api').toggle_linewise_op(vim.fn.visualmode())<CR>")
 end
 
-M.lspconfig = function()
+M.lspconfig = function(bufnr)
    -- See `<cmd> :help vim.lsp.*` for documentation on any of the below functions
-   map("n", "gD", function()
+
+   local buf_map = function(...)
+      local key, lhs, rhs = ...
+      map(key, lhs, rhs, { buffer = bufnr })
+   end
```

Comment:
This could be in the utils file but I don't think it is used anywhere else + it needs some domifications